### ISSUE TITLE
DCS protocol does not send last frame

### ISSUE THREAD

**Issue by yo2loj** [LX3JL/xlxd#13]
On outgoing DCS streams, the last frame is not encoded as a last frame.
Cause: incorrect conditions in cdcsprotocol.cpp, method HandleQueue:

```
            if ( packet->IsDvFrame() )
            {
                EncodeDvPacket(m_DvHeadersCache[iModId], (const CDvFramePacket &)*packet, m_iSeqCounters[iModId]++, &buffer);
            }
            else if ( packet->IsLastPacket() )
            {
                EncodeDvLastPacket(m_DvHeadersCache[iModId], (const CDvFramePacket &)*packet, m_iSeqCounters[iModId]++, &buffer);
            }

```
Last packet is a DvFrame, so the else condition will never be reached.

Corrected code:
```
            if ( packet->IsDvFrame() )
            {
                if ( packet->IsLastPacket() )
                {
                    EncodeDvLastPacket(m_DvHeadersCache[iModId], (const CDvFramePacket &)*packet, m_iSeqCounters[iModId]++, &buffer);
                }
                else
                {
                    EncodeDvPacket(m_DvHeadersCache[iModId], (const CDvFramePacket &)*packet, m_iSeqCounters[iModId]++, &buffer);
                }
            }
```
Patch file attached.

[cdcsprotocol.diff.zip](https://github.com/LX3JL/xlxd/files/419060/cdcsprotocol.diff.zip)

**Comment by LX3JL** [comment]
Hi Marius,

You are correct.
I will incorporate your suggestion in next release, coming today or tomorow.
Thank you for your help

73
JL

**Comment by yo2loj** [comment]
Solved in 1.3.8
